approvals/approvals/api.py

```diff
@@ -165,13 +168,36 @@ def set_status_to_approved(doc: Document, method: str | None = None, automatic=F
 
 
 @frappe.whitelist()
-def reject_document(doc: Document, role=None, comment: str = "", method: str | None = None):
+def reject_document(
+	doc: Document,
+	role=None,
+	comment: str = "",
+	document_approval_rule_name: str = "",
+	method: str | None = None,
+):
+	from approvals.approvals.doctype.document_approval_rule.document_approval_rule import get_users
+
 	doc = frappe._dict(json.loads(doc)) if isinstance(doc, str) else doc
 	doc = frappe.get_doc(doc.doctype, doc.name)
 	doc.save(ignore_permissions=True)
 	doc.set_status(update=True, status="Rejected")
 	rejection = add_comment(doc.doctype, doc.name, comment, frappe.session.user, frappe.session.user)
 	revoke_approvals_on_reject(doc, method)
+
+	if document_approval_rule_name:
+		document_approval_rule = frappe.get_doc("Document Approval Rule", document_approval_rule_name)
+	else:
+		document_approval_rule = frappe.new_doc("Document Approval Rule")
```

Comment:
When there isn't a `Document Approval Rule` I create a new object without saving it to the DB only for the sake of using the `assign_user` method.